### ISSUE TITLE
auto-resolved promises fire after the test ends

### ISSUE THREAD

**Issue by dfalling** [emberjs/ember-mocha#126]
The test:
```js
  it('displays the result', function() {
    // setup
    this.promise = Ember.RSVP.resolve('success');
    let serviceStub = Ember.Service.extend({ search: () => this.promise });
    this.register('service:asyncService', serviceStub);
    this.inject.service('asyncService', { as: 'asyncService' });

    //run
    this.render(hbs`{{async-component}}`);
    this.$('#searchInput').val('search term', 'trigger', 'input');
    this.$('#searchButton').click();

    //assertions
    expect(this.$('#result').text()).to.contain('success');
  });
```

async-component.hbs:
```hbs
{{input id="searchInput" value=searchText}}
<button id="searchButton" {{action "search" searchText}}>Search</button>
<div id="result">result: {{result}}</div>
```

async-component.js:
```js
import Ember from 'ember';

export default Ember.Component.extend({
  asyncService: Ember.inject.service(),
  result: undefined,

  actions: {
    async search(term) {
      let result = await this.get('asyncService').search(term);
      this.set('result', result);
    }
  }
});
```

A similar test in QUnit works fine, not requiring any special handling for async, since the promise is set to auto-resolve.  In mocha the assertions run and fail, then the promise resolves.  I've noticed several other issues involving component tests, especially [#54](https://github.com/emberjs/ember-mocha/issues/54).  Is there any progress being made to fix this?  I can provide the entire test project if that would help.

Thanks!

**Comment by Turbo87** [comment]
@dfalling I might be misinterpreting this, but I think we are handling this correctly. `Promise.resolve()` is supposed to not resolve synchronously, but instead will resolve async. That means that if you're asserting directly after the `click()` call, the Promise won't be resolved yet.

The solution to your problem can be found at https://guides.emberjs.com/v2.10.0/testing/testing-components/#toc_waiting-on-asynchronous-behavior

tl;dr:

```js
return wait().then(() => {
  expect(this.$('#result').text()).to.contain('success');
});
```

or since you seem to be using async/await:

```js
await wait();
expect(this.$('#result').text()).to.contain('success');
```
